components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -737,21 +794,46 @@ private boolean isDomainNameEmbeddedInAttributeValue(String filterOperation, Str
      * @return User names of the filtered users
      * @throws CharonException Error while filtering
      */
-    private String[] getFilteredUsernames(Node node, int offset, int limit, String sortBy, String sortOrder,
+    private String[] getFilteredUsernames(Condition condition, int offset, int limit, String sortBy, String sortOrder,
             String domainName) throws CharonException {
 
-        if (log.isDebugEnabled())
+        if (log.isDebugEnabled()) {
             log.debug("Filtering users in domain : " + domainName);
-
+        }
         try {
-            Map<String, String> attributes = getAllAttributes(domainName);
             return ((PaginatedUserStoreManager) carbonUM)
-                    .getUserList(getCondition(node, attributes), domainName, UserCoreConstants.DEFAULT_PROFILE, limit,
-                            offset, sortBy, sortOrder);
+                    .getUserList(condition, domainName, UserCoreConstants.DEFAULT_PROFILE, limit, offset, sortBy,
+                            sortOrder);
+        } catch (UserStoreException e) {
+            String errorMessage = String
+                    .format("Error while retrieving users for the domain: %s with limit: %d and " + "offset: %d.",
```

Comment:
remove the unwanted " + " in the middle of the string

Reply:
fixed: dc26889c2156cbd827075fa28e80cddd63dc87c4